### PR TITLE
[fga] Skip old permission system if `centralizedPermissions` is enabled

### PR DESCRIPTION
**Pull request by geropl**
## Description
This introduces `FGAResourceAccessGuard` which skips the old permission system if `centralizedPermissions` is enabled for the given user. To that end it cleans up some of the old `guardAccess*` methods, and unifies usage, code patterns and feature-flag check.

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at cbf3d22</samp>

Refactored and cleaned up server code related to centralized permissions feature. Removed deprecated metrics code, added new access control method, and fixed user lookup logic.

</details>

## Related Issue(s)

Fixes EXP-445, EXP-449

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - gpl-fga-skip-access</li>
	<li><b>🔗 URL</b> - <a href="https://gpl-fga-skip-access.preview.gitpod-dev.com/workspaces" target="_blank">gpl-fga-skip-access.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - gpl-fga-skip-access-gha.15446</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Comment by geropl** [comment]
Let's re-base and merge after this PR: https://github.com/gitpod-io/gitpod/pull/18517

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 1]
```diff
@@ -592,7 +617,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             throw new ApplicationError(ErrorCodes.NOT_AUTHENTICATED, "User is not authenticated. Please login.");
         }
 
-        const user = await this.userService.findUserById(this.userID, this.userID);
+        const user = await this.userService.findUserById(SYSTEM_USER, this.userID);
```
Comment (svenefftinge) [inline components/server/src/workspace/gitpod-server-impl.ts:620]: why do you treat this as a system call? I think this is only called based on a user action.

**Review by svenefftinge — COMMENTED** [review]
Why did you introduce a new method instead of checking the enablement in `guardAccess` directly? Also it only disables only half of the calls.

File: components/server/src/workspace/gitpod-server-impl.ts (hunk 1 above)
Comment (geropl) [reply components/server/src/workspace/gitpod-server-impl.ts:620]: I see `checkUser` as something that we, the system, want to perform to validate the user is actually allowed to interact with it, and enforce other invariants. This should be independent of any permissions the user has or not has.

**Comment by geropl** [comment]
> Why did you introduce a new method instead of checking the enablement in guardAccess directly? Also it only disables only half of the calls.

To selectively only disable access checks for which we already have a FGA replacement:
 1. it helps with making that change incremental
 2. we can start testing before we fully implement all changes

File: components/server/src/auth/resource-access.ts [hunk 2]
```diff
@@ -152,6 +154,26 @@ export class CompositeResourceAccessGuard implements ResourceAccessGuard {
     }
 }
 
+/**
+ * FGAResourceAccessGuard can disable the delegate if FGA is enabled.
+ */
+export class FGAResourceAccessGuard implements ResourceAccessGuard {
+    constructor(private readonly userId: string, private readonly delegate: ResourceAccessGuard) {}
+
+    async canAccess(resource: GuardedResource, operation: ResourceAccessOp): Promise<boolean> {
```
Comment (geropl) [inline components/server/src/auth/resource-access.ts:163]: @svenefftinge Used your idea with the `ResourceGuard`, so headless-log is covered as well ([see below](https://github.com/gitpod-io/gitpod/pull/18519/files#diff-ad3bbabbf5bd475f8c522ec6cba0b0745ce8260a8989064f5feb6bb9e9a4b4d8R228)).

**Review by svenefftinge — APPROVED** [review]
code LGTM

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 3]
```diff
@@ -2457,72 +2453,38 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
...
-            if (
-                await getExperimentsClientForBackend().getValueAsync("centralizedPermissions", false, {
-                    teamId: org.id,
-                })
-            ) {
-                reportCentralizedPermsValidation(fineGrainedOp, fromDB.value === fromCentralizedPerms.value);
+    // TODO(gpl) Remove after FGA rollout
+    private async hasOrgOperationPermission(org: Organization, members: TeamMemberInfo[], op: ResourceAccessOp) {
+        try {
+            await this.guardAccess({ kind: "team", subject: org, members }, op);
```
[the remark this answers is not in the document]
Comment (geropl) [reply components/server/src/workspace/gitpod-server-impl.ts]: Nice find. Got change-blind here. :100:

**Pull request merged**